src/modules/claims/claims.types.ts

```diff
@@ -96,3 +99,201 @@ export const defaultClaimExpiry = Number.MAX_SAFE_INTEGER - 1; // constraint of
 export type RequestClaim = { requester: string; message: IClaimRequest };
 export type IssueClaim = { issuer: string; message: IClaimIssuance };
 export type RejectClaim = { issuer: string; message: IClaimRejection };
+
+export interface GetClaimsByRequesterOptions {
+  /** DID of the requestor */
+  did: string;
+
+  /** Indicates whether to show only accepted `Claims`  */
+  isAccepted?: boolean;
+
+  /** Indicates what namespace `Claims` should be in  */
+  namespace?: string;
+}
+
+export interface GetClaimsByIssuerOptions {
+  /** DID of the issuer */
+  did: string;
+
+  /** Indicates whether to show only accepted `Claims` */
+  isAccepted?: boolean;
+
+  /** Indicates what namespace `Claims` should be in */
+  namespace?: string;
+}
+
+export interface GetClaimsBySubjectOptions {
+  /** DID of the subject */
+  did: string;
+
+  /** Indicates whether to show only accepted `Claims` */
+  isAccepted?: boolean;
+
+  /** Indicates what namespace `Claims` should be in */
+  namespace?: string;
+}
+
+export interface CreateClaimRequestOptions {
+  /** Claim request params */
+  claim: {
+    /** Role namespace */
+    claimType: string;
+
+    /** Version of the role */
+    claimTypeVersion: number;
+
+    /** Requestor fields that role is requiring */
+    requestorFields?: { key: string; value: string | number }[];
+  };
+
+  /** DID of the subject */
+  subject?: string;
+
+  /** Indicates what type of claim registration you are requesting: on-chain and/or off-chain */
+  registrationTypes?: RegistrationTypes[];
+}
+
+export interface IssueClaimRequestOptions {
+  /** DID of the claim requestor */
+  requester: string;
+
+  /** JWT token generated by requestor during claim request */
+  token: string;
+
+  /** Claim id */
+  id: string;
+
+  /** Subject agreement signature */
+  subjectAgreement: string;
+
+  /** Registration types */
+  registrationTypes: RegistrationTypes[];
+
+  /** Issuer fields that role is requiring */
+  issuerFields?: { key: string; value: string | number }[];
+
+  /** Indicates whether to publish role on-chain or not (default: false) */
+  publishOnChain?: boolean;
+}
+
+export interface RegisterOnchainOptions {
+  /** Role namespace */
+  claimType?: string;
+
+  /** Version of the claim type */
+  claimTypeVersion?: string;
+
+  /** @deprecated */
+  token?: string;
+
+  /** Subject agreement signature */
+  subjectAgreement?: string;
+
+  /** On-chain proof signature */
+  onChainProof: string;
+
+  /** DID of the issuer */
+  acceptedBy: string;
+
+  /** DID of the claim subject */
+  subject?: string;
+}
+
+export interface RejectClaimRequestOptions {
+  /** Claim id */
+  id: string;
+
+  /** DID of the claim requestor */
+  requesterDID: string;
+
+  /** Reason for rejection */
+  rejectionReason?: string;
+}
+
+export interface DeleteClaimOptions {
```

Comment:
What you mean? Having not inline types have couple of pros: someone can import type (impossible with inline types), more readable code and less code in main service file.

Reply:
If parameter is primitive type like `id` then it can be imported, is more readable then `DeleteClaimOptions` and less code

Reply:
How it can be imported? You can do `import {DeleteClaimOptions} from 'iam-client-lib'`, but how to import inline type without `ReturnType<typeof ClaimService['deleteClaim']>`?

Reply:
Why to import it? It is a string

Reply:
It's a object that contain one property with string type, it's not a string. I didn't change it. https://github.com/energywebfoundation/iam-client-lib/blob/57489056a4c2111eabea869b72f3a31f73c197d3/src/modules/claims/claims.service.ts#L435

Reply:
Ah, ok then